test/validation/api/time/time.c

```diff
@@ -765,19 +765,41 @@ static void time_test_global_sync(const int ctrl)
 	CU_ASSERT(odph_thread_join(thread_tbl, num) == num);
 
 	for (int s = 0; s < TIME_SAMPLES; s++) {
+		int min_idx = 0, max_idx = 0;
 		uint64_t min = UINT64_MAX, max = 0;
+		double avg = 0;
 
 		for (int i = 1; i < num + 1; i++) {
 			uint64_t t = odp_time_to_ns(global_mem->time[i][s]);
 
-			if (t < min)
+			if (t < min) {
 				min = t;
-			if (t > max)
+				min_idx = i;
+			}
+		}
+
+		printf("\nround %d\nthread time diffs: ", s);
+
+		for (int i = 1; i < num + 1; i++) {
+			uint64_t t = odp_time_to_ns(global_mem->time[i][s]) - min;
+
+			printf("%" PRIu64 " ", t);
+
+			if (t > max) {
 				max = t;
+				max_idx = i;
+			}
+
+			avg += t;
 		}
 
-		CU_ASSERT(max - min < tolerance);
+		avg /= num - 1;
```

Comment:
Avg should be divided by "num". The for loop runs "num" times.

Reply:
It's num - 1 because min result itself is not included in the average. Comment added in v3.